sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -730,4 +730,11 @@ class DataFrameSuite extends QueryTest {
     val res11 = ctx.range(-1).select("id")
     assert(res11.count == 0)
   }
+
+  test("SPARK-8621: support empty string column name") {
+    val df = Seq(Tuple1(1)).toDF("").as("t")
+    // We should allow empty string as column name
+    df.col("")
+    df.col("t.``")
```

Comment:
This line is also giving some error , mainly "AnalysisException: cannot resolve 't.``' given input columns ;"